practice_app/api/models.py

```diff
@@ -1,3 +1,57 @@
 from django.db import models
+from django.contrib.auth.models import User
+
+import json
 
 # Create your models here.
+
+class Paper(models.Model):
+    paper_id = models.AutoField(primary_key=True)
+    third_party_id = models.CharField(max_length=100)
+    source = models.CharField(max_length=50)
+    abstract = models.TextField(max_length=5000, blank=True, null=True)
+    year = models.IntegerField(blank=True, null=True)
+    url = models.URLField()
+    authors = models.TextField(max_length=500, blank=True, null=True)
+    title = models.CharField(max_length=100)
+    like_count = models.IntegerField()
+
+    def set_authors(self, authors):
+        self.authors = json.dumps(authors)
+
+    def get_authors(self):
+        return json.loads(self.authors) if self.authors else []
+
+class Comment(models.Model):
+    # comment_id = models.AutoField(primary_key=True)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+    paper =  models.ForeignKey(Paper, on_delete=models.CASCADE)
+    text = models.TextField(max_length=500)
+
+class Like(models.Model):
```

Comment:
Shouldn't combination of the user and paper be unique? I think composit keys should be used here.

Reply:
You're right. I added a unique constraint for those keys.